### PR TITLE
wildan3105/github-langs#23: Add screenshot testing to application

### PR DESCRIPTION
**Pull request by metamaker**
Changes:

See Readme.md changes to have to run screenshot testing process. Resulting screenshots are located inside `tests/screenshot-testing/screenshots/`

### CODE REVIEW

**Comment by metamaker** [comment]
@wildan3105 finally ready :) This was interesting experience and now great feature of this project.

**Comment by wildan3105** [comment]
Wow! Thanks @metamaker, I'll review this tomorrow when I had the chance :smile:, please be patient and thanks again, great work :100:

**Comment by metamaker** [comment]
:+1: You can clone my branch from https://github.com/metamaker/github-langs/tree/patch-phantom-1 to verify locally how it works and play around ;)

**Comment by wildan3105** [comment]
Just curious @metamaker, If we made a change on the result page like add `share` and `tweet` button, does it affect the testing method? If it does, which part should we change and how could we make this process as modular as possible? Sorry for questioning like this because I'm totally new to `screenshot testing`. Thanks!

**Comment by metamaker** [comment]
@wildan3105 

1. You run test according to instruction in README.
2. If nothing is broken, you will have 

screen shot 2017-10-13 at 17 44 57

Drink smoothie, no significant change happened, so test passed.

3. If some change happened, you will have

screen shot 2017-10-13 at 17 56 58

In folder `tests/screenshot-testing/screenshots` you will have `.diff.png` files that contain screenshot of how site looks NOW, just `.png` contain screenshot of how site looked before changes. If you know that some change will break look of site, just remove `.png` screenshot from that folder - `npm test` will automatically create new for you and think that it is ok.

> If we made a change on the result page like add share and tweet button, does it affect the testing method?

You don't need to write any additional code, just delete screenshots of pages that changed in `tests/screenshot-testing/screenshots` and run `npm test` to recreate screenshots.

Only screenshots are verified to be equal. List of screenshots to take is located in https://github.com/wildan3105/github-langs/pull/52/files#diff-fc0dea3ad35358fa472f2a00bd9f9cec (suite.js) file.

Keep latest screenshots in `master` branch. This way you can prevent regressions. Plus, you will have newest screenshots of your app as free bonus 😄 .

See also https://www.sitepoint.com/visual-regression-testing-with-phantomcss/, https://github.com/Huddle/PhantomCSS.

**Comment by wildan3105** [comment]
Ah I see! Thanks for the comprehensive explanation @metamaker ! Good job :+1:

File: tests/screenshot-testing/suite.js [hunk 1]
```diff
@@ -0,0 +1,106 @@
+var fs = require('fs');
```
Comment (wildan3105) [inline tests/screenshot-testing/suite.js]: Could you make the `var` to be `const`?

File: tests/screenshot-testing/suite.js [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+				red: 255,
+				green: 255,
+				blue: 0
+			},
+			errorType: 'movement',
+			transparency: 0.3
+		}*/
+	});
+
+	casper.on('remote.message', function (msg) {
```
Comment (wildan3105) [inline tests/screenshot-testing/suite.js]: And use `arrow function` for all `funcs`

**Comment by wildan3105** [comment]
Hi @metamaker could you resolve the conflict and change what I have requested? Thank you very much :)

**Comment by metamaker** [comment]
Ok.

**Comment by metamaker** [comment]
@wildan3105 done

**Comment by wildan3105** [comment]
Thanks! @metamaker

**Pull request merged**